test/sqs_lambda_to_batch_fargate/lambda/export_logs.py

```diff
@@ -0,0 +1,81 @@
+import json
+import logging
+import os
+import time
+
+import boto3
+
+s3_client = boto3.client("s3")
+s3_resource = boto3.resource("s3")
+logs_client = boto3.client("logs")
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def handler(event, context):
+    logger.info(f"INCOMING EVENT: {event}")
+    if "Batch Job State Change" in event["detail-type"]:
+        if (
+            "FAILED" in event["detail"]["status"]
+            or "SUCCEEDED" in event["detail"]["status"]
+        ):
+            logger.info(f"BUCKET_NAME: {os.environ['BUCKET_NAME']}")
+            try:
+                log_group_name = "/aws/batch/job"
+
+                # Get most recent stream
+                log_streams = logs_client.describe_log_streams(
+                    logGroupName=log_group_name,
+                    orderBy="LastEventTime",
+                    descending=True,
+                    limit=1,
+                )
+
+                # Get log events from the stream
+                log_events = logs_client.get_log_events(
+                    logGroupName=log_group_name,
+                    logStreamName=log_streams["logStreams"][0]["logStreamName"],
+                    startFromHead=True,
+                )
+
+                # Generate a timestamp for the current time
+                timestamp_str = int(time.time())
+                file_name = f"{timestamp_str}.log"
+                with open(f"/tmp/{file_name}", "w") as output_file:
+                    for event in log_events["events"]:
+                        log_entry = f"[{event['timestamp']}], {event['message']}\n"
+                        output_file.write(log_entry)
+
+                s3_client.upload_file(
+                    f"/tmp/{file_name}", os.environ["BUCKET_NAME"], file_name
+                )
+
+                s3_client.put_object(
+                    Body=f"/tmp/{file_name}",
+                    Bucket=os.environ["PRODUCER_BUCKET_NAME"],
+                    Key=f"{os.environ['LANGUAGE_NAME']}/{file_name}",
+                )
+
+                logger.info(f"Log data saved successfully: {file_name}")
+
+                return {
```

Comment:
Here you `return {statusCode: 200}`, elsewhere you `return {statusCode: 500}`, and elsewhere you `return` (None), and elsewhere you `raise`. As we talked on slack, I don't see where the preferred approach is documented? So putting on my "how would I write it" hat, any return is a success and any raise is an error. Since this is responding to SQS, the http-isms aren't helpful. Therefore, I'm going to suggest that the successful cases should be a logger.info and a `return None`, and the two exceptional cases are a `rase Exception`

Reply:
After more research, this is a "must" (and as written, it doesn't do what it looks like it does)